Lib/test/test_clinic.py

```diff
@@ -19,17 +20,14 @@
     from clinic import DSLParser
 
 
-class _ParserBase(TestCase):
-    maxDiff = None
-
-    def expect_parser_failure(self, parser, _input):
-        with support.captured_stdout() as stdout:
-            with self.assertRaises(SystemExit):
-                parser(_input)
-        return stdout.getvalue()
-
-    def parse_function_should_fail(self, _input):
-        return self.expect_parser_failure(self.parse_function, _input)
+def _expect_failure(tc, parser, code, errmsg, *, filename=None, lineno=None):
+    _code = dedent(code).strip()
+    with tc.assertRaisesRegex(clinic.ClinicError, errmsg) as cm:
```

Comment:
What if we had a call to `re.escape` here? It would mean we wouldn't have to have all the `re.escape` calls in all the individual test methods:

```suggestion
    errmsg = re.escape(errmsg)
    with tc.assertRaisesRegex(clinic.ClinicError, errmsg) as cm:
```

Reply:
Doh!

Reply:
That's what you get for late night coding 😆